src/dmd/target.d

```diff
@@ -61,12 +63,13 @@ import dmd.root.string : toDString;
 extern (C++) struct Target
 {
     // D ABI
-    uint ptrsize;             /// size of a pointer in bytes
-    uint realsize;            /// size a real consumes in memory
-    uint realpad;             /// padding added to the CPU real size to bring it up to realsize
-    uint realalignsize;       /// alignment for reals
-    uint classinfosize;       /// size of `ClassInfo`
-    ulong maxStaticDataSize;  /// maximum size of static data
+    uint ptrsize;                   /// size of a pointer in bytes
+    uint realsize;                  /// size a real consumes in memory
+    uint realpad;                   /// padding added to the CPU real size to bring it up to realsize
+    uint realalignsize;             /// alignment for reals
+    uint classinfosize;             /// size of `ClassInfo`
+    ulong maxStaticDataSize;        /// maximum size of static data
+    Platform platform;              /// backend platform
```

Comment:
I think this field is redundant?

Reply:
Yes at the moment as currently `globals` need to be initialized beforehand, but I thought you wanted to move this into `target.d`.
I can also remove the field until it's possible to use it solely.

Reply:
Ideally this information should be private to target.d.  The idea being that at the language level, we should not care about what CPU or OS we are compiling for.  This means that conditions should be rephrased into Target(C|CXX) hooks or fields.

**Don't ask**: Is the target I'm compiling for one of Linux, OSX, FreeBSD, OpenBSD, DragonFlyBSD, or Solaris?
**Do ask**: Does the target I'm compiling for have void C++ constructors?

Reply:
(https://issues.dlang.org/show_bug.cgi?id=12327)